roles/semaphore_deployment/templates/deploy.j2

```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
+
+branch=$@
+allowed=^[a-zA-Z0-9._/\-]+$
+
+# Given branch name must be a single string with no spaces,
+# containing only: ".", "-", "/", "_", or alphanumeric characters.
+
+if [[ ! $branch =~ $allowed ]]
+then
+  echo "Invalid branch name given."
+  exit;
```

Comment:
```suggestion
  exit 1
```
That indicates that the script failed. You don't need `;` if you have a newline at the end.